File: 447-number-of-boomerangs.cs
public class Solution {
    public int NumberOfBoomerangs(int[][] points) {
                int res = 0;
                foreach (var point in points) {
                    var dict = new Dictionary<int, int>();
                    foreach (var targetPoint in points) {
                        int dx = point[0] - targetPoint[0];
                        int dy = point[1] - targetPoint[1];
                        var dis = dx * dx + dy * dy;
                        dict[dis] = dict.GetValueOrDefault(dis, 0) + 1;
                    }
                    foreach (var(key, value) in dict) {
                        res += (value * (value - 1));
                    }
                }
                return res;
    }
}

File: 447-number-of-boomerangs.go
package main

func numberOfBoomerangs(points [][]int) int {
	ret := 0
	for _, p := range points {
		dict := map[int]int{}
		for _, q := range points {
			dis := (p[0]-q[0])*(p[0]-q[0]) + (p[1]-q[1])*(p[1]-q[1])
			dict[dis]++
		}

		for _, m := range dict {
			ret += m * (m - 1)
		}
	}
	return ret
}
